batch/batches/Convert/Engines/KJobConversionEngine.class.php

```diff
@@ -146,15 +148,125 @@ public function convertJob ( KalturaConvertJobData &$data )
 		
 		return array ( true , $error_message );// indicate all was converted properly
 	}
-	
+
+	protected function isConversionProgressing($currentModificationTime)
+	{
+		if (kFile::checkFileExists($this->inFilePath))
+		{
+			$newModificationTime = kFile::getFileLastUpdatedTime($this->inFilePath);
+			if ($newModificationTime !== false && $newModificationTime > $currentModificationTime)
+			{
+				return $newModificationTime;
+			}
+		}
+		return false;
+	}
+
+	protected function getReturnValues($handle)
+	{
+		$return_var = 1;
+		$output = false;
+		if ($handle)
+		{
+			$return_var = 0;
+			$file = $this->outFilePath;
+			if (kFile::checkFileExists($file))
+			{
+				$output = kFile::getLineFromFileTail($file , 1);
+			}
+		}
+		KalturaLog::debug('getReturnValues, output is: '. $output. ' return var is '. $return_var);
+		return array($output, $return_var);
+	}
+
 	/**
 	 *
 	 */
-	protected function execute_conversion_cmdline($command, &$return_var)
+	protected function execute_conversion_cmdline($command, &$return_var, $data = null)
```

Comment:
The new code should be executed only if the worker data includes parameter that indicates - usingSmartJobTimeout=1.
Otherwise call the function with the old code. if the values is set - call a function with your new code